### DISCUSSION TITLE
Speeding up compile times with LLD or mold linker

### DISCUSSION THREAD

**Discussion by RaoulLuque** [bevyengine/bevy#11903] [Q&A]
As explained at the bottom of the [setup page](https://bevyengine.org/learn/quick-start/getting-started/setup/) in the bevy getting started guide, using either LLD or mold as a linker should increase the compile times (drastically)?

I have tried this for [this](https://github.com/RaoulLuque/rust-invasion) little test game that I have written and doing a fresh cargo run (after deleting the target folder) gives me a compile time of 8 minutes and an even slower compile time of 8 minutes and 30 seconds when using LLD.

For this I have run
``` sudo apt-get install lld ```
as suggested for ubuntu users and added the following to my .cargo/config.toml file:
```
[target.x86_64-pc-windows-msvc]
rustflags = ["-C", "link-arg=-fuse-ld=lld"]

[target.x86_64-pc-windows-gnu]
rustflags = ["-C", "link-arg=-fuse-ld=lld"]

[target.x86_64-unknown-linux-gnu]
rustflags = ["-C", "linker=clang", "-C", "link-arg=-fuse-ld=lld"]
```

For that I also had to install clang with this command (since it wouldn't compile otherwise so I guess it actually used LLD at least):
``` sudo apt install clang ```

Maybe it's just the project that doesn't seem to benefit from LLD or am I doing something wrong here? Also 8 minutes and 30 seconds seems like quite a long time for a fresh compile. Of course the re-compiles after changes in my code take about 5/10 seconds so that's really fine.

I have tried the same with the mold linker and added the following to my .cargo/config.toml instead:
``` [target.x86_64-unknown-linux-gnu]
linker = "clang"
rustflags = ["-C", "link-arg=-fuse-ld=/usr/bin/mold"]
```

Additionally as explained in the Mold repo I have checked if the binary is actually linked with mold and it is indeed so that shouldn't be the issue:
```
String dump of section '.comment':
  [     0]  mold 1.0.3 (compatible with GNU ld)
  [    25]  GCC: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
  [    50]  rustc version 1.76.0 (07dca489a 2024-02-04)
```

**Comment by Rexcrazy804** [comment]
I don't have any experience with mold, but I am certain that LLD and MOLD are intended to make recompiles (after changes in your code) marginally faster. On my system with LLD the recompiles are roughly 5-10 seconds faster.

I've went through the setup page you've linked this is what was mentioned: 

> Bevy can be built just fine using default configuration on stable Rust. However, for **maximally fast iterative compiles**, we recommend the following configuration:

emphasis on "maximally fast **iterative compiles**". So yeah, it's meant to make recompiles (using your terminology) faster rather than making the initial compilation faster.
